### ISSUE TITLE
Feature request: Switch RoutingEngines/MapRendering in two multitasking options

### ISSUE THREAD

**Issue by Caballlero** [rinigus/osmscout-server#147]
I was thinking that it would be good to use SF multitasking options, in this piece of gold. just sharing ideas!

**Comment by rinigus** [comment]
Actually, with Valhalla, you will be able to calculate route, render in Mapnik, and search in Geocoder-NLP all in parallel. The server does keep track and runs tasks in parallel using as many CPUs as you have. My phone has 4, so 4 things can be done in parallel when I use Mapnik/Valhalla/Geocoder-NLP. 

So, if I do understand correctly, its already done :)

PS: Its a right way to propose ideas, thank you very much!

**Comment by Caballlero** [comment]
Then what about the dual core-jolla 1 ?

**Comment by Caballlero** [comment]
rinigus My main question was,on a icon/option in multitask to switch between mapnik/valhalla and osmscout/osmscout.

**Comment by rinigus** [comment]
Re Jolla 1: yes, then it should run 2 processes at the same time. Its a slower phone => slower response.

Re cover: I see, never thought that interaction via covers would be called multitasking :)

Indeed, cover is used only for giving the information on the server state (how many jobs does it do and whether it downloads). 

When we discussed Valhalla integration (see corresponding issue in github), it was suggested to focus on providing one and well-designed option. In this sense, it would be better to keep such straightforward approach. Having extra buttons for switching configuration may lead to more confusion, I think. Also note, when you switch between Mapnik and Osmscout, you have to clean up cache of Poor Maps/modRana. All these complications make "simple" switch maybe a bit oversimplification.

Ideally, I would like to get the server working so that people just fire it up and don't have to change anything. Since you suggested it, do you change backends frequently?

I am preparing a release now with new functionality. After that I am planning to mainly polish the server GUI. So, let's think what we want from the server and see how to make it easy and intuitive to use. 

For now, let's keep this issue open to have a discussion on how to use cover better and what kind of interaction users would want.

**Comment by Caballlero** [comment]
I call  "multitask button" to "interaction via covers"too. Because this feature is multifunctional.  :  )
I know that multitasking, in an operating system is allowing a user to perform more than one computer task . I apologize rinigus.
- Yes you are right about cached of  Maps/modRana.
- So what function would be best for you if we use an option in interaction via covers ?

**Comment by rinigus** [comment]
Re Function via covers:

Actually, I don't know what to put there. In some sense it goes against "start and forget" principle that I am trying to implement on the server side. I guess, as soon as settings are fine, users would need server's GUI only for downloading/updating/removing maps. So, unless we get any "killer" feature idea via such interaction, I think it is fine to have cover just showing the status.

PS: no worries :)

**Comment by Caballlero** [comment]
yes RE: Button via covers = ) jiji

**Comment by Caballlero** [comment]
Will close this, in reference to you idea.

**Comment by rinigus** [comment]
Caballlero, thank you very much for your input! Let's keep this issue closed and see what ideas we'll get in future!
